backend/src/services/kms/kms-service.ts

```diff
@@ -66,22 +66,55 @@ export const kmsServiceFactory = ({
 }: TKmsServiceFactoryDep) => {
   let ROOT_ENCRYPTION_KEY = Buffer.alloc(0);
 
+  const getRootEncryptionKey = (encryptionAlgorithm: SymmetricEncryption) => {
+    switch (encryptionAlgorithm) {
+      case SymmetricEncryption.AES_GCM_128:
+        return ROOT_ENCRYPTION_KEY.subarray(0, 16); // taking first 128bits
+      case SymmetricEncryption.AES_GCM_256:
+      default:
+        return ROOT_ENCRYPTION_KEY;
+    }
+  };
+
   /*
    * Generate KMS Key
    * This function is responsibile for generating the infisical internal KMS for various entities
    * Like for secret manager, cert manager or for organization
    */
-  const generateKmsKey = async ({ orgId, isReserved = true, tx, slug }: TGenerateKMSDTO) => {
-    const cipher = symmetricCipherService(SymmetricEncryption.AES_GCM_256);
-    const kmsKeyMaterial = randomSecureBytes(32);
-    const encryptedKeyMaterial = cipher.encrypt(kmsKeyMaterial, ROOT_ENCRYPTION_KEY);
+  const generateKmsKey = async ({
+    orgId,
+    isReserved = true,
+    tx,
+    slug,
+    projectId,
+    encryptionAlgorithm = SymmetricEncryption.AES_GCM_256,
+    description
+  }: TGenerateKMSDTO) => {
+    const cipher = symmetricCipherService(encryptionAlgorithm);
+
+    let bytes: number;
```

Comment:
Its better to move this to a separate function outside the service. As these will be quite heavily used utility functions going forward

Reply:
refactored